backend/functions/src/middleware/auth-middleware.ts

```diff
@@ -23,3 +41,39 @@ export function checkAuth(req: Request, res: Response, next: NextFunction) {
     res.status(401).send('Unauthorized: please sign in')
   }
 }
+
+export function checkIsAuthorized(
+  req: Request,
+  res: Response,
+  next: NextFunction
+) {
+  const idToken = req.headers?.authorization?.split('Bearer ')[1]
+  if (idToken) {
+    admin
+      .auth()
+      .verifyIdToken(idToken)
+      .then((decodedToken) => {
+        const uid = decodedToken.uid
+        admin
+          .auth()
+          .getUser(uid)
+          .then((user) => {
+            if (user.email) {
```

Comment:
could these nested if-else statements be simplified into just: 
'''
if(user.email && allowedUsers.includes(user.email)) {
    next()
} else {
    res.status(403).send('Unauthorized: not correct permissions')
}
'''